apps/aecore/lib/aecore/naming/name_util.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Aecore.Naming.NameUtil do
+  @moduledoc """
+  Aecore naming utilities.
+  """
+
+  alias Aeutil.Hash
+
+  @split_name_symbol "."
+
+  @name_registrars [@split_name_symbol <> "aet", @split_name_symbol <> "test"]
+
+  @spec normalized_namehash!(String.t()) :: binary()
+  def normalized_namehash!(name) do
+    case normalize_and_validate_name(name) do
+      {:ok, normalized_name} -> namehash(normalized_name)
+      {:error, error} -> throw(error)
+    end
+  end
+
+  @spec normalize_and_validate_name(String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  def normalize_and_validate_name(name) do
+    normalized_name = normalize_name(name)
+
+    case validate_normalized_name(normalized_name) do
+      :ok -> {:ok, normalized_name}
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  @spec normalize_name(String.t()) :: String.t()
+  def normalize_name(name),
+    do:
+      name
+      |> :idna.utf8_to_ascii()
+      |> to_string()
+
+  @spec namehash(String.t()) :: binary()
+  defp namehash(name) do
+    if name == "" do
+      <<0::256>>
+    else
+      {label, remainder} = partition_name(name)
+      Hash.hash(namehash(remainder) <> Hash.hash(label))
+    end
+  end
+
+  @spec partition_name(String.t()) :: {String.t(), String.t()}
+  defp partition_name(name) do
+    [label | remainder] = split_name(name)
+    {label, Enum.join(remainder, @split_name_symbol)}
+  end
+
+  @spec validate_normalized_name(String.t()) :: :ok | {:error, String.t()}
+  defp validate_normalized_name(name) do
+    allowed_registrar =
+      Enum.any?(@name_registrars, fn registrar -> String.ends_with?(name, registrar) end)
+
+    if allowed_registrar do
+      validate_name_length(name)
+    else
+      {:error, "name doesn't end with allowed registrar"}
+    end
+  end
+
+  @spec validate_name_length(String.t()) :: :ok | {:error, String.t()}
+  defp validate_name_length(name) do
+    case String.length(name) > 0 && String.length(name) < 253 do
```

Comment:
Shouldn't this max length name value be stored somewhere and not "hardcoded" here?